javascript/javascript1/week2/index.js

```diff
@@ -0,0 +1,215 @@
+    *********************************************/ 
+
+
+const class07Students = [];
+let count=0;
+function addStudentToClass(studentName) { 
+
+    if(class07Students.includes(studentName)){
+            console.log("Student " +studentName+ "  is already in the class");
+            
+    }else if(count>=7 && studentName=="Margrethe")  {
+        class07Students.push(studentName);
+        // console.log(class07Students);
+
+    }
+      else if(count<7){
+               class07Students.push(studentName);
+              // console.log(class07Students);
+               count++;
+    }
+    else
+                console.log('Cannot add more students to class 07');
+
+}
+addStudentToClass("name1");
+addStudentToClass("name2");
+addStudentToClass("name3");
+addStudentToClass("name4");
+addStudentToClass("name5");
+addStudentToClass("name6");
+addStudentToClass("name7");
+addStudentToClass("Margrethe");    // Queen is added. Even if the class has been filled out
+addStudentToClass("name2");        // loged out:  Student name2  is already in the class
+addStudentToClass("name8");        // loged out : Cannot add more students to class 07
+
+
+
+
+function getNumberOfStudents() {
+  class07Students.push("name9","name10","name11");   // Add some students to the class.
+  class07Students.push("name12","name13","name14");  // Add more students than there is space for
+  class07Students.push("name1","name2","name3");     //Add a student that is already in the class
+  class07Students.push("Margrethe");                    //Add the Queen to a full class
+  //console.log(class07Students);
+   return class07Students.length;
+}
+let numberOfStudents=getNumberOfStudents();
+console.log("Total number of students in class = "+numberOfStudents);// log out the number of students =18
+
+
+console.log('\n\n  5 :  Candy helper   \n\n ');
+
+
+    /******************************************** 
+    *   5   :    Candy helper                    *
+    *********************************************/
+
+
+let boughtCandyPrices = [];
+let candyPrice='';
+function addCandy(candyType,weight){
+ 
+    if(candyType=='sweet'){
+        candyPrice=0.5*weight;
+}else if(candyType=='chocolate'){
+        candyPrice=0.7*weight;
+}else if (candyType=='toffe'){
+        candyPrice=1.1*weight;
+}else if(candyType=='chewing-gum'){
+        candyPrice=0.03*weight;
+}
+ boughtCandyPrices.push(candyPrice);
+// console.log(boughtCandyPrices);
+//console.log(candyPrice);
+ 
+}
+addCandy("sweet", 20); 
+addCandy("chocolate", 20);
+addCandy("toffe", 20);
+addCandy("chewing-gum", 20);
+
+   let amountToSpend=Math.random() * 100;
+
+
+// console.log(amountToSpend);
+
+function canBuyMoreCandy() {
+    let sum =0;
+    for(let i=0; i<boughtCandyPrices.length; i++){
+        sum= sum + boughtCandyPrices[i];
+        }
+        //console.log(sum);
+        if (sum<=amountToSpend){
+            return true
+
+        }else
+        return false;
+
+}
+let c=canBuyMoreCandy();
```

Comment:
try to use the variable name more meaningfull

Reply:
sorry for this .. did not noticed it i changed it to const buyMore